File: kotlin/gigasecond/src/main/kotlin/Gigasecond.kt
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.temporal.Temporal
import java.time.temporal.TemporalUnit

class Gigasecond {
  var date: LocalDateTime? = null
  val billion: Long = 1_000_000_000

  constructor(date: LocalDate) {
    this.date = date.atStartOfDay().plusSeconds(billion)
  }

  constructor(date: LocalDateTime) {
    this.date = date.plusSeconds(billion)
  }


}

File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond
  (:import [java.time LocalDateTime]))


(def gigasecond 1000000000)


(defn date->result [date]
  [(.getYear date) (-> date .getMonth .getValue) (.getDayOfMonth date)])


(defn from [year month day]
  (-> (LocalDateTime/of year month day 0 0)
      (.plusSeconds gigasecond)
      (date->result)))
